public/app/features/explore/AddToDashboard/addToDashboard.ts

```diff
@@ -25,13 +24,25 @@ const createDashboard = (
   return getDashboardSrv().saveDashboard({ dashboard, folderId }, { showErrorAlert: false, showSuccessAlert: false });
 };
 
-export const addToDashboard = async (data: SaveToNewDashboardDTO, options: SaveOptions): Promise<string> => {
-  const res = await createDashboard(
-    data.dashboardName,
-    data.folderId,
-    options.queries,
-    options.datasource,
-    options.panel
-  );
-  return res.data.url;
+const addPanelToDashboard = async (uid: string, queries: DataQuery[], datasource: DataSourceRef, panel: string) => {
+  const dashboardData = await backendSrv.getDashboardByUid(uid);
+
+  const dashboard = new DashboardModel(dashboardData.dashboard, dashboardData.meta);
+
+  dashboard.addPanel({ targets: queries, type: panel, title: 'New Panel', datasource });
+
+  return getDashboardSrv().saveDashboard({ dashboard }, { showErrorAlert: false, showSuccessAlert: false });
```

Comment:
Can you test this with a dashboard with variables? it's possible this is going to remove all the variables from the dashboard, or it might be that it will get all the variables from the previously opened dashboard. As saving a dashboard will take the current variables from redux.

Reply:
good point, i didn't think of that use case. there is also an issue with doubled app notifications when creating a panel that i still need to tackle (hence why this is till in draft)